### PR TITLE
[SPARK-10026] [ML] [PySpark] Implement some common Params for regression in PySpark

### PR DESCRIPTION
**Pull request by yanboliang**
LinearRegression and LogisticRegression lack of some Params for Python, and some Params are not shared classes which lead we need to write them for each class. These kinds of Params are list here:

``` scala
HasElasticNetParam 
HasFitIntercept
HasStandardization
HasThresholds
```

Here we implement them in shared params at Python side and make LinearRegression/LogisticRegression parameters peer with Scala one.

### CODE REVIEW

File: python/pyspark/ml/classification.py [hunk 1]
```diff
@@ -31,7 +31,8 @@
 
 @inherit_doc
 class LogisticRegression(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPredictionCol, HasMaxIter,
-                         HasRegParam, HasTol, HasProbabilityCol, HasRawPredictionCol):
+                         HasRegParam, HasTol, HasProbabilityCol, HasRawPredictionCol,
+                         HasElasticNetParam, HasFitIntercept, HasStandardization, HasThresholds):
```
Comment (feynmanliang) [inline python/pyspark/ml/classification.py]: `HasThresholds` is for multi-class but `ml.LogisticRegression` currently only supports binary classification. Perhaps we should also extract a `HasThreshold` mixin for binary classifier thresholds

File: python/pyspark/ml/classification.py [hunk 2]
```diff
@@ -65,17 +66,6 @@ class LogisticRegression(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPredicti
...
-              "the ElasticNet mixing parameter, in range [0, 1]. For alpha = 0, " +
-              "the penalty is an L2 penalty. For alpha = 1, it is an L1 penalty.")
-    fitIntercept = Param(Params._dummy(), "fitIntercept", "whether to fit an intercept term.")
-    thresholds = Param(Params._dummy(), "thresholds",
-                       "Thresholds in multi-class classification" +
-                       " to adjust the probability of predicting each class." +
-                       " Array must have length equal to the number of classes, with values >= 0." +
-                       " The class with largest value p/t is predicted, where p is the original" +
-                       " probability of that class and t is the class' threshold.")
     threshold = Param(Params._dummy(), "threshold",
```
Comment (feynmanliang) [inline python/pyspark/ml/classification.py]: Perhaps we should also extract a `HasThreshold` mixin for binary classifier thresholds

File: python/pyspark/ml/param/shared.py [hunk 3]
```diff
@@ -444,7 +555,7 @@ class DecisionTreeParams(Params):
     minInfoGain = Param(Params._dummy(), "minInfoGain", "Minimum information gain for a split to be considered at a tree node.")
     maxMemoryInMB = Param(Params._dummy(), "maxMemoryInMB", "Maximum memory in MB allocated to histogram aggregation.")
     cacheNodeIds = Param(Params._dummy(), "cacheNodeIds", "If false, the algorithm will pass trees to executors to match instances with nodes. If true, the algorithm will cache node IDs for each instance. Caching can speed up training of deeper trees.")
-
+
```
Comment (feynmanliang) [inline python/pyspark/ml/param/shared.py]: Remove whitespace

File: python/pyspark/ml/param/shared.py [hunk 4]
```diff
@@ -460,7 +571,7 @@ def __init__(self):
         self.maxMemoryInMB = Param(self, "maxMemoryInMB", "Maximum memory in MB allocated to histogram aggregation.")
         #: param for If false, the algorithm will pass trees to executors to match instances with nodes. If true, the algorithm will cache node IDs for each instance. Caching can speed up training of deeper trees.
         self.cacheNodeIds = Param(self, "cacheNodeIds", "If false, the algorithm will pass trees to executors to match instances with nodes. If true, the algorithm will cache node IDs for each instance. Caching can speed up training of deeper trees.")
-
+
```
Comment (feynmanliang) [inline python/pyspark/ml/param/shared.py]: Ditto

**Comment by feynmanliang** [comment]
LGTM :+1:, some minor formatting comments and a suggestion.

File: python/pyspark/ml/classification.py (hunk 2 above)
Comment (yanboliang) [inline python/pyspark/ml/classification.py]: ```threshold``` is a deprecated parameter, and it is replaced by ```thresholds```. LogisticRegression still reserve ```threshold``` is just for binary compatibility. So I think we don't need to extract ```HasThreshold``` as shared Param.

Comment (feynmanliang) [inline python/pyspark/ml/classification.py]: My understanding is that the `HasThresholds` trait mixin in `ml.LogisticRegression` is actually an artifact resulting from transient dependency through `ProbabilisticClassifier`. We don't actually support multi-class classification in `ml.LogisticRegression` ATM and did quite a bit of work to make the API less confusing.

After mutli-class is supported I think it makes sense to use `HasThresholds`, but for the time being I would prefer we only use `HasThreshold` in the Python API.

File: python/pyspark/ml/classification.py [hunk 5]
```diff
@@ -124,13 +97,13 @@ def __init__(self, featuresCol="features", labelCol="label", predictionCol="pred
...
-                  threshold=0.5, thresholds=None,
-                  probabilityCol="probability", rawPredictionCol="rawPrediction"):
+                  threshold=0.5, thresholds=None, probabilityCol="probability",
+                  rawPredictionCol="rawPrediction", standardization=True):
         """
         setParams(self, featuresCol="features", labelCol="label", predictionCol="prediction", \
                   maxIter=100, regParam=0.1, elasticNetParam=0.0, tol=1e-6, fitIntercept=True, \
-                  threshold=0.5, thresholds=None, \
-                  probabilityCol="probability", rawPredictionCol="rawPrediction")
+                  threshold=0.5, thresholds=None, probabilityCol="probability",
```
Comment (mengxr) [inline python/pyspark/ml/classification.py]: add `\`

File: python/pyspark/ml/regression.py [hunk 6]
```diff
@@ -63,38 +64,30 @@ class LinearRegression(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPrediction
...
-
     @keyword_only
     def __init__(self, featuresCol="features", labelCol="label", predictionCol="prediction",
-                 maxIter=100, regParam=0.0, elasticNetParam=0.0, tol=1e-6):
+                 maxIter=100, regParam=0.0, elasticNetParam=0.0, tol=1e-6, fitIntercept=True,
+                 standardization=True):
         """
         __init__(self, featuresCol="features", labelCol="label", predictionCol="prediction", \
-                 maxIter=100, regParam=0.0, elasticNetParam=0.0, tol=1e-6)
+                 maxIter=100, regParam=0.0, elasticNetParam=0.0, tol=1e-6, fitIntercept=True,
```
Comment (mengxr) [inline python/pyspark/ml/regression.py]: add `\`

File: python/pyspark/ml/regression.py [hunk 7]
```diff
@@ -63,38 +64,30 @@ class LinearRegression(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPrediction
...
 
     @keyword_only
     def setParams(self, featuresCol="features", labelCol="label", predictionCol="prediction",
-                  maxIter=100, regParam=0.0, elasticNetParam=0.0, tol=1e-6):
+                  maxIter=100, regParam=0.0, elasticNetParam=0.0, tol=1e-6, fitIntercept=True,
+                  standardization=True):
         """
         setParams(self, featuresCol="features", labelCol="label", predictionCol="prediction", \
-                  maxIter=100, regParam=0.0, elasticNetParam=0.0, tol=1e-6)
+                  maxIter=100, regParam=0.0, elasticNetParam=0.0, tol=1e-6, fitIntercept=True,
```
Comment (mengxr) [inline python/pyspark/ml/regression.py]: add `\`

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**